src/main/scala/codacy/tailor/Tailor.scala

```diff
@@ -35,69 +30,95 @@ object Tailor extends Tool {
-        .map(name => Try(File(source.path) / name))
-        .collectFirst {
-          case Success(file) if file.isRegularFile =>
-            List("-c", file.toJava.getAbsolutePath)
-        }
+  private def patternsToUseFromCodacyConfig(
+      configuration: Option[List[Pattern.Definition]]
+  )(implicit specification: Specification): Option[List[String]] = {
+    val patternsToLintOpt = configuration.withDefaultParameters
+
+    patternsToLintOpt.map {
+      case patternsToLint if patternsToLint.nonEmpty =>
+        val patternIds = patternsToLint.map(_.patternId)
+        val parameters =
+          patternsToLint.flatMap(_.parameters).flatten.flatMap { parameter =>
+            List(s"--${parameter.name}", paramValueToJsValue(parameter.value).toString)
+          }
+        List("--only=" + patternIds.mkString(",")) ++ parameters
+      case _ => List.empty[String]
+    }
+  }
 
-      val filesToLint: List[String] = files.fold(List(source.path)) { paths =>
-        paths.map(_.toString).toList
-      }
+  private def lintingConfiguration(source: Source.Directory, configuration: Option[List[Pattern.Definition]])(
+      implicit specification: Specification
+  ): List[String] = {
+    lazy val nativeConfig = nativeConfigurationFromSource(source)
 
-      val patternsToLintOpt = configuration.withDefaultParameters
-
-      val conf: Option[List[String]] = patternsToLintOpt.map {
-        case patternsToLint if patternsToLint.nonEmpty =>
-          val patternIds = patternsToLint.map(_.patternId)
-          val parameters =
-            patternsToLint.flatMap(_.parameters).flatten.flatMap { parameter =>
-              List(s"--${parameter.name}",
-                   paramValueToJsValue(parameter.value).toString)
-            }
-          List("--only=" + patternIds.mkString(",")) ++ parameters
-        case _ => List.empty[String]
-      }
+    patternsToUseFromCodacyConfig(configuration)
+      .orElse(nativeConfig)
+      .getOrElse(List.empty)
+  }
 
-      val cfgOpt = conf.orElse(nativeConfig).getOrElse(List.empty)
-
-      val command = List("/usr/bin/tailor/bin/tailor", "-f", "json") ++ cfgOpt ++ List(
-        "--") ++ filesToLint
-
-      CommandRunner.exec(command) match {
-        case Right(resultFromTool) =>
-          parseToolResult(Paths.get(source.path), resultFromTool.stdout) match {
-            case s @ Success(_) => s
-            case Failure(e) =>
-              val msg =
-                s"""
-                   |${this.getClass.getSimpleName} exited with code ${resultFromTool.exitCode}
-                   |command: ${command.mkString(" ")}
-                   |message: ${e.getMessage}
-                   |stdout: ${resultFromTool.stdout.mkString(
-                     Properties.lineSeparator)}
-                   |stderr: ${resultFromTool.stderr.mkString(
-                     Properties.lineSeparator)}
+  private def failureMessage(command: List[String], resultFromTool: CommandResult, e: Throwable): String = {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [failureMessage should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)